### PR TITLE
[SMT] Add constant boolean operation

### PR DESCRIPTION
**Pull request by maerhart**

### CODE REVIEW

**Review by TaoBi22 — APPROVED** [review]
LGTM!

**Review by fabianschuiki — APPROVED** [review]
LGTM!

File: include/circt/Dialect/SMT/SMTOps.td [hunk 1]
```diff
@@ -49,6 +49,25 @@ def DeclareConstOp : SMTOp<"declare_const", [
...
+  let description = [{
+    Produces the constant expressions 'true' and 'false' as described in the
+    [Core theory](https://smtlib.cs.uiowa.edu/Theories/Core.smt2) of the SMT-LIB
+    Standard 2.6.
+  }];
+
+  let arguments = (ins BoolAttr:$value);
+  let results = (outs BoolType:$result);
+  let assemblyFormat = "$value attr-dict";
+
```
Comment (math-fehr) [inline include/circt/Dialect/SMT/SMTOps.td:67]: You'll probably at some point want a builder that just take a boolean, but maybe that's already generated (?)

**Review by math-fehr — APPROVED** [review]
Cool!

File: include/circt/Dialect/SMT/SMTOps.td (hunk 1 above)
Comment (maerhart) [reply include/circt/Dialect/SMT/SMTOps.td:67]: Had the same thought, so I checked, and it's actually already auto-generated 👍

**Pull request merged**